spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java

```diff
@@ -128,7 +128,8 @@ public String description() {
 			if (this.properties.getMetadata() != null && this.properties.getMetadata().isAddPorts()) {
 				subset.getPorts().stream().forEach(p -> metadata.put(p.getName(), Integer.toString(p.getPort())));
 			}
-			V1EndpointPort port = subset.getPorts() != null && subset.getPorts().size() == 1 ? subset.getPorts().get(0)
+			V1EndpointPort port = subset.getPorts() != null &&
+				(subset.getPorts().size() == 1 || this.properties.getPrimaryPortName() == null) ? subset.getPorts().get(0)
```

Comment:
I dont think this logic makes sense.  If there are multiple ports they will need names, so I think it would be safe to change the logic slightly to avoid the potential NPE 

`p.getName().equasIgnoreCase(this.properties.getPrimaryPortName())`

Reply:
Okay, that would work.  
The other problem is the code get exceptions when there are multiple ports and none of them matches with primaryPortName. I spent half an hours to figure it out. It should be better to provide more information in the log.

Reply:
> I dont think this logic makes sense. If there are multiple ports they will need names, so I think it would be safe to change the logic slightly to avoid the potential NPE
> 
> `p.getName().equasIgnoreCase(this.properties.getPrimaryPortName())`

Okay, that would work.
The other problem is the code get exceptions when there are multiple ports and none of them matches with primaryPortName. I spent half an hours to figure it out. It should be better to provide more information in the log.

Reply:
@zhaozhihom feel free to add some logging and/or documentation around that problem as well